src/store/RelayContext.js

```diff
@@ -0,0 +1,279 @@
+/**
+ * Copyright 2013-2015, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule RelayContext
+ * @typechecks
+ * @flow
+ */
+
+'use strict';
+
+var GraphQLFragmentPointer = require('GraphQLFragmentPointer');
+var GraphQLStoreQueryResolver = require('GraphQLStoreQueryResolver');
+import type RelayMutation from 'RelayMutation';
+var RelayDeferredFragmentState = require('RelayDeferredFragmentState');
+import type {
+  RelayDeferredFragmentStateOptions
+} from 'RelayDeferredFragmentState';
+var RelayMutationTransaction = require('RelayMutationTransaction');
+var RelayQuery = require('RelayQuery');
+var RelayQueryResultObservable = require('RelayQueryResultObservable');
+var RelayStoreData = require('RelayStoreData');
+
+var forEachRootCallArg = require('forEachRootCallArg');
+var readRelayQueryData = require('readRelayQueryData');
+
+import type {
+  Abortable,
+  Observable,
+  RelayMutationTransactionCommitCallbacks,
+  ReadyStateChangeCallback,
+  StoreReaderData,
+  StoreReaderOptions
+} from 'RelayTypes';
+
+import type {
+  DataID,
+  RelayQuerySet
+} from 'RelayInternalTypes';
+
+/**
+ * @public
+ *
+ * RelayStore is a caching layer that records GraphQL response data and enables
+ * resolving and subscribing to queries.
+ *
+ * === onReadyStateChange ===
+ *
+ * Whenever Relay sends a request for data via GraphQL, an "onReadyStateChange"
+ * callback can be supplied. This callback is called one or more times with a
+ * `readyState` object with the following properties:
+ *
+ *   aborted: Whether the request was aborted.
+ *   done: Whether all response data has been fetched.
+ *   error: An error in the event of a failure, or null if none.
+ *   ready: Whether the queries are at least partially resolvable.
+ *   stale: When resolvable during `forceFetch`, whether data is stale.
+ *
+ * If the callback is invoked with `aborted`, `done`, or a non-null `error`, the
+ * callback will never be called again. Example usage:
+ *
+ *  function onReadyStateChange(readyState) {
+ *    if (readyState.aborted) {
+ *      // Request was aborted.
+ *    } else if (readyState.error) {
+ *      // Failure occurred.
+ *    } else if (readyState.ready) {
+ *      // Queries are at least partially resolvable.
+ *      if (readyState.done) {
+ *        // Queries are completely resolvable.
+ *      }
+ *    }
+ *  }
+ *
+ */
+class RelayContext {
+  _storeData: RelayStoreData;
+
+  constructor(storeData?: RelayStoreData) {
+    this._storeData = storeData || new RelayStoreData();
```

Comment:
I don't really like exposing this in the constructor, but it seemed like the lesser evil for removing the `_getStoreData` calls that the tests required. I think ideally the tests wouldn't require so much mocking, but that would require more refactoring...